anndata/_core/views.py

```diff
@@ -157,6 +164,11 @@ def as_view_zappy(z, view_args):
     return z
 
 
+@as_view.register(AwkArray)
+def as_view_awkarray(array, view_args):
+    return AwkwardArrayView(array, view_args=view_args)
```

Comment:
@ivirshup, I have no clue what's going on here: 

* `as_view_awkarray` is correctly called when accessing e.g. `adata.obsm["awk"]` when adata is a view, but
* it returns an `ak.Array`, and not an instance of `AwkwardArrayView`
* therefore, it doesn't have the `__setitem__` inherited from `_ViewMixin`, and doesn't copy on modifcation. 

```python
>>> from anndata import AnnData
>>> import anndata
>>> import numpy as np
>>> import awkward._v2 as ak

>>> adata = AnnData(np.full((4,4), 1))
>>> adata.obsm["awk"] = ak.Array([{"a": 2}] * 4)
>>> v = adata[:3,:3]
>>> v
View of AnnData object with n_obs × n_vars = 3 × 3
   obsm: 'awk'

>>> v.obsm['awk']
<Array [{a: 2}, {a: 2}, {a: 2}] type='3 * {a: int64}'>

>>> isinstance(v.obsm['awk'], anndata._core.views.AwkwardArrayView)
False
```

Reply:
I [tried that](https://github.com/scverse/anndata/pull/647/files#diff-18c0605ea4c7c29d7151bc4297ce90d5e0e5e1435f5db76ec943f3c51c0d188aR160-R176), but I'm unsure what the best way is to pass the `view_args`.

Reply:
Asked: https://github.com/scikit-hep/awkward/discussions/1177#discussioncomment-3535483

Reply:
@ivirshup, I implemented the approach with weakref, as discussed in https://github.com/scikit-hep/awkward/discussions/1177. The `test_view` (in `test_awkward.py`) passes now. Please LMK what you think.